lib/internal/modules/cjs/loader.js

```diff
@@ -1024,26 +1036,9 @@ Module._extensions['.node'] = function(module, filename) {
   return process.dlopen(module, path.toNamespacedPath(filename));
 };
 
-Module._extensions['.mjs'] = function(module, filename) {
-  throw new ERR_REQUIRE_ESM(filename);
-};
-
 // Bootstrap main module.
-Module.runMain = function() {
-  // Load the main module--the command line argument.
-  if (experimentalModules) {
-    asyncESM.loaderPromise.then((loader) => {
-      return loader.import(pathToFileURL(process.argv[1]).href);
-    })
-    .catch((e) => {
-      internalBinding('errors').triggerUncaughtException(
-        e,
-        true /* fromPromise */
-      );
-    });
-    return;
-  }
-  Module._load(process.argv[1], null, true);
+Module.runMain = function(main) {
```

Comment:
Since this is public...we should probably run CITGM for this and make sure e.g. esm is fine with it.

Reply:
I can keep the bootstrap in this file if you would prefer? Was kind of just following your refactoring comment in moving it out, but there's no reason we have to do that.

Reply:
I've kept the full runMain here, calling out to the separated functions.